File: Assignment 1/Problem11.py
def findPeakElement(nums):
    left, right = 0, len(nums) - 1

    while left <= right:
        mid = left + (right - left) // 2

        if (mid == 0 or nums[mid] > nums[mid - 1]) and (mid == len(nums) - 1 or nums[mid] > nums[mid + 1]):
            return mid
        elif mid < len(nums) - 1 and nums[mid] < nums[mid + 1]:
            left = mid + 1
        else:
            right = mid - 1

    return -1

nums = [1, 2, 3, 1]
peak_index = findPeakElement(nums)
print(peak_index)

File: ShrutiVarshney_BanasthaliVidyapith_ArraysAssignment/ArraysAssignmentQuestions/src/arraysassignmentquestions/Problem11.java
package arraysassignmentquestions;
import java.util.*;
public class Problem11 
{
    public int findPeakElement(int[] nums) 
    {
        int left = 0;
        int right = nums.length - 1;
        
        while (left < right) 
        {
            int mid = left + (right - left) / 2;
            
            if (nums[mid] < nums[mid + 1]) 
            {
                // Peak is on the right side
                left = mid + 1;
            } 
            else 
            {
                // Peak is on the left side or at the mid index
                right = mid;
            }
        }
        
        return left;
    }
    public static void main(String[] args) 
    {
        Scanner sc=new Scanner(System.in);
        System.out.println("Enter the length of the array 'nums' - 'n':");
        int n=sc.nextInt();
        int[] nums=new int[n];
        for(int i=0;i<n;i++)
        {
            System.out.println("Enter the element"+(i+1));
            nums[i]=sc.nextInt();
        }

        Problem11 solution = new Problem11();
        int peakIndex = solution.findPeakElement(nums);

        // Print the peak index
        System.out.println("Peak Index: " + peakIndex);
    }
}
